src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionItemData.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.VisualStudio.Language.Intellisense.AsyncCompletion.Data;
+using Microsoft.VisualStudio.Text;
+using RoslynCompletionItem = Microsoft.CodeAnalysis.Completion.CompletionItem;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.AsyncCompletion
+{
+    internal sealed class CompletionItemData
+    {
+        private const string RoslynCompletionItemData = nameof(RoslynCompletionItemData);
+
+        public RoslynCompletionItem RoslynItem { get; }
+
+        public Optional<SnapshotPoint> TriggerLocation { get; }
+
+        public bool IsProvidedByRoslynCompletionSource => TriggerLocation.HasValue;
+
+        private CompletionItemData(RoslynCompletionItem roslynItem, SnapshotPoint? triggerLocation)
+        {
+            RoslynItem = roslynItem;
+            TriggerLocation = triggerLocation ?? new();
```

Comment:
❕ Don't use `new T()` (or `new()`) for value types unless they have a default constructor. Also, I'm concerned this line may be treating `new()` as `new SnapshotPoint()` and not `new Optional<SnapshotPoint>()` (even if it's not, it's confusing).

Reply:
You are right, new() here actually is `new SnapshotPoint()`. Fixed